api/cloudcontroller/wrapper/uaa_authentication.go

```diff
@@ -86,3 +77,24 @@ func (t *UAAAuthentication) Wrap(innerconnection cloudcontroller.Connection) clo
 	t.connection = innerconnection
 	return t
 }
+
+// refreshToken refreshes the JWT access token if it is expired or about to expire.
+// If the access token is not yet expired, no action is performed.
+func (t *UAAAuthentication) refreshToken() error {
+	expiration := time.Time{}
+	tokenStr := strings.TrimPrefix(t.cache.AccessToken(), "bearer ")
+	token, err := jws.ParseJWT([]byte(tokenStr))
+	if err == nil {
+		// if the JWT could not be parsed, the empty value of time.Time forces a refresh
+		expiration, _ = token.Claims().Expiration()
```

Comment:
`token.Claims().Expiration()` is not an option, if the token cannot be parsed in the first place since `ParseJWT()` returns nil. While adding a zero valued token into the scope would solve this particular problem, it would just hide the problem of the zero value time even further.

Instead I made the duration until the token expires more explicit (`expiresIn time.Duration`). I hope this makes the code easier to read.

Reply:
Ah I see what you're saying. I think I misread the logic when making this comment. 

The changes definitely help clarify the behavior.